barretenberg/cpp/src/barretenberg/vm/avm/trace/trace.cpp

```diff
@@ -3197,23 +3196,39 @@ AvmError AvmTraceBuilder::op_get_contract_instance(
         error = AvmError::CHECK_TAG_ERROR;
     }
 
-    // Read the contract instance
-    ContractInstanceHint instance = execution_hints.contract_instance_hints.at(read_address.val);
-
-    FF member_value;
-    switch (chosen_member) {
-    case ContractInstanceMember::DEPLOYER:
-        member_value = instance.deployer_addr;
-        break;
-    case ContractInstanceMember::CLASS_ID:
-        member_value = instance.contract_class_id;
-        break;
-    case ContractInstanceMember::INIT_HASH:
-        member_value = instance.initialisation_hash;
-        break;
-    default:
-        member_value = 0;
-        break;
+    FF member_value = 0;
+    bool exists = false;
+
+    if (is_ok(error)) {
+        // Read the contract instance
+        ContractInstanceHint instance = execution_hints.contract_instance_hints.at(read_address.val);
+        // nullifier read hint for the contract address
+        NullifierReadTreeHint nullifier_read_hint = instance.membership_hint;
+        // hinted nullifier should match the specified contract addrss
+        exists = nullifier_read_hint.low_leaf_preimage.nullifier == read_address.val;
+        ASSERT(exists);
```

Comment:
I don't think we should assert here? If `read_address.val` doesn't exist in the tree that this equality would be false?